pkg/monitor/monitor/util.go

```diff
@@ -406,7 +406,13 @@ func getMonitorDMInitContainer(monitor *v1alpha1.TidbMonitor, dc *v1alpha1.DMClu
 }
 
 func getMonitorPrometheusContainer(monitor *v1alpha1.TidbMonitor, tc *v1alpha1.TidbCluster) core.Container {
-	commands := []string{"sed 's/$NAMESPACE/'\"$NAMESPACE\"'/g;s/$POD_NAME/'\"$POD_NAME\"'/g' /etc/prometheus/config/prometheus.yml > /etc/prometheus/config_out/prometheus.yml && /bin/prometheus --web.enable-admin-api --web.enable-lifecycle --config.file=/etc/prometheus/config_out/prometheus.yml --storage.tsdb.path=/data/prometheus " + fmt.Sprintf("--storage.tsdb.retention=%dd", monitor.Spec.Prometheus.ReserveDays)}
+	var retention string
+	if monitor.Spec.Prometheus.RetentionTime != nil {
```

Comment:
```suggestion
	if monitor.Spec.Prometheus.RetentionTime != nil && *monitor.Spec.Prometheus.RetentionTime != "" {
```

Reply:
RetentionTime have validated, it can't be ""